week-4/E-arrays-of-objects/exercise-3.js

```diff
@@ -60,14 +60,19 @@ var restaurantFinderApplication = {
     applicationVersion: "1.0",
     restaurants: restaurants,
     findAvailableRestaurants: function(numberOfPeople) {
-        // Complete here
-    },
+
+        var resAvailable = restaurants.filter(rest=>
+             (rest.totalSeats-rest.numberOfCustomers>numberOfPeople)).map(dest=>dest.name);
+        return resAvailable;
+         },
     findRestaurantServingDish: function(dishName) {
-        // Complete here
+        var dishInResturant = restaurants.filter(dish=>dish.menu.includes(dishName)).map(rest=>rest.name);
+        return dishInResturant;
     },
     countNumberOfRestaurantsInArea: function(area) {
-        // Complete here
-    }
+     var restInGlasgow  = restaurants.filter(gArea=>gArea.address.area === area).length;
+     return restInGlasgow;
+      }
```

Comment:
Great job - this runs smoothly and returns what we want. 👏

Like before, I think we are putting too much on one line here. And also, we have some issues with variable names. 

- `dish` isn't ideal because you then write: `dish.menu`, which doesn't make sense. A menu has dishes, a dish doesn't have a menu. At that point in time, you'll dealing with an array of restaurants even if you objective is to reference the dish eventually. 
- `gArea` - what does this mean? I guess short for glasgow area? I would avoid short acronyms that might not be obvious to other developers working with your code. Better be verbose and obvious than short and obscure.

I don't want to sound like I am repeating myself, but these are important things to think about. When you start programming with other developers, they will need to be able to understand what is going on with your code 😊 That being said, the code works great! 👍